inst/support.hpp

```diff
@@ -56,84 +60,83 @@ void user_check_array_value(const std::vector<T>& value, const char *name,
   }
 }
 
-inline size_t user_get_array_rank(Rcpp::RObject x) {
-  if (x.hasAttribute("dim")) {
-    Rcpp::IntegerVector dim = x.attr("dim");
-    return dim.size();
-  } else {
-    // This is not actually super correct
+inline size_t user_get_array_rank(cpp11::sexp x) {
+  if (!::Rf_isArray(x)) {
```

Comment:
This gets `Rf_isArray` from the global namespace? Just trying to understand why that is. So this is in `Rinternals.h`? Which I guess is pulled in by one of the cpp11 includes? Is it not callable via just `Rf_isArray`? How is this different to like `R_NilValue` which you call from e.g. https://github.com/mrc-ide/dust/pull/39/files#diff-b76e188a9cd3bc5135b37e6b22d651a2R64

Reply:
Ah and R_NilValue in this file https://github.com/mrc-ide/odin.dust/pull/14/files#diff-6f14a897fd7f2eaaf15453eeb5a6f703R124

Reply:
There's no real need for ::Rf_isArray at all - cpp11 does not seem to have any helper for "is this SEXP an array" so I've just grabbed the underlying C function. R_NilValue is the sentinel value for R's `NULL` (**very** different to C/C++'s `NULL`!). They will come in via `R.h` which cpp11 is bringing in.